lib/ruby_indexer/lib/ruby_indexer/visitor.rb

```diff
@@ -113,9 +108,11 @@ def collect_comments(node)
 
     sig { params(name: String).returns(String) }
     def fully_qualify_name(name)
-      return name if @stack.empty?
-
-      "#{@stack.join("::")}::#{name}"
+      if @stack.empty? || name.start_with?("::")
+        name
+      else
+        "#{@stack.join("::")}::#{name}"
+      end.delete_prefix("::")
```

Comment:
We can avoid the `name.start_with?("::")` since we're always deleting the prefix anyway.
```suggestion
      normalized_name = name.delete_prefix("::")
      return normalized_name if @stack.empty?

      "#{@stack.join("::")}::#{normalized_name}"
```

Reply:
The current code already always deletes the prefix. `name.start_with?("::")` is to avoid prepending `@stack`, not for the deletion.

Reply:
Sorry, I don't understand. If it always deletes the prefix, when would `name.start_with?("::")` match?